spec/lib/common/client/middleware/request/remove_cookies_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+describe Common::Client::Middleware::Request::RemoveCookies do
+  class TestConfiguration < Common::Client::Configuration::REST
+    def port
+      3010
+    end
+
+    def connection
+      @conn ||= Faraday.new("http://127.0.0.1:#{port}") do |faraday|
+        faraday.use(Common::Client::Middleware::Request::RemoveCookies)
+        faraday.adapter :httpclient
+      end
+    end
+  end
+
+  class TestService < Common::Client::Base
+    configuration TestConfiguration
+  end
+
+  describe '#request' do
+    let!(:server_thread) do
+      Thread.new do
+        server = WEBrick::HTTPServer.new(
+          Port: TestConfiguration.instance.port
+        )
+
+        server.mount_proc '/' do |req, res|
+          res.header['Set-Cookie'] = 'foo=bar'
+          res.body = req.header['cookie'].to_json
+        end
+
+        server.start
+      end
+    end
+
+    it 'should strip cookies' do
+      VCR.configure do |c|
+        c.allow_http_connections_when_no_cassette = true
+      end
+
+      Timeout.timeout(5) do
+        loop do
```

Comment:
What's the reasoning behind not just sending two requests?

Reply:
the thread is asynchronous so if this isn't done then it fails on the first request because the server hasn't booted up yet